boa/src/environment/declarative_environment_record.rs

```diff
@@ -73,32 +75,45 @@ impl EnvironmentRecordTrait for DeclarativeEnvironmentRecord {
                 strict,
             },
         );
-
-        true
+        Ok(())
     }
 
-    fn initialize_binding(&mut self, name: &str, value: Value) {
+    fn initialize_binding(&mut self, name: &str, value: Value) -> Result<(), ErrorKind> {
         if let Some(ref mut record) = self.env_rec.get_mut(name) {
             if record.value.is_none() {
                 record.value = Some(value);
+                Ok(())
             } else {
                 // TODO: change this when error handling comes into play
-                panic!("Identifier {} has already been defined", name);
+                Err(ErrorKind::ReferenceError(format!(
+                    "Identifier {} has already been defined",
+                    name
+                )))
             }
+        } else {
+            panic!(format!("record must have binding for {}", name));
         }
     }
 
     #[allow(clippy::else_if_without_else)]
-    fn set_mutable_binding(&mut self, name: &str, value: Value, mut strict: bool) {
+    fn set_mutable_binding(
+        &mut self,
+        name: &str,
+        value: Value,
+        mut strict: bool,
+    ) -> Result<(), ErrorKind> {
         if self.env_rec.get(name).is_none() {
             if strict {
                 // TODO: change this when error handling comes into play
-                panic!("Reference Error: Cannot set mutable binding for {}", name);
+                return Err(ErrorKind::ReferenceError(format!(
+                    "Reference Error: Cannot set mutable binding for {}",
+                    name
+                )));
             }
 
             self.create_mutable_binding(name.to_owned(), true);
```

Comment:
```suggestion
            self.create_mutable_binding(name.to_owned(), true)?;
```
The error should be propagated with `?` operator.

Reply:
Yes. Thanks for pointing out. Actually I've ignored the compiler warnings for now since I just wanted to know from you how to deal with errors in the `exec` module.

Reply:
In exec, one possible way is to call the method on environment and further call `.or_else` on the Result type to map `ErrorKind` to an actual error object.

```Rust
env.create_mutable_binding(
    param.to_owned(),
    false,
    VariableScope::Block,
 ).or_else(|e| Err(e.to_error(interpreter)))?;
```